### ISSUE TITLE
[feature] Project templates

### ISSUE THREAD

**Issue by con-f-use** [mitsuhiko/rye#87]
In larger organizations, it is often useful to have some kind of cookie cutter or template for starting new python projects. A popular solution for python projects would be the [Cookiecutter](https://github.com/cookiecutter/cookiecutter) project. I was even thinking of some kind of mechanism, where something like django could define a template and rye would be able to list that (and others) as a possible project starter if django (or whatever) is already installed. Sort of an entrypoint-system or plugin system for project templates. Just an idea. You could also just wrap Cookiecutter, or leave it out completely. Packaging just has a lot of biolerplate, that a unified tool could take care of.

**Comment by cnpryer** [comment]
I'd love for `rye` to have opinions. If `rye` has opinions here less is probably better. This does sound like plugin territory. I see this like `rye` supporting any other *default*. If there is a default it probably should be pretty minimal. I'm a fan of the src layout.

Here's an example project of mine:
```zsh
xlcsv on  master is 📦 v0.1.0 via 🐍 v3.11.0 
❯ tree .
.
├── LICENSE
├── Makefile
├── README.md
├── pyproject.toml
├── requirements-dev.lock
├── requirements.lock
├── src
│   └── xlcsv
│       ├── __init__.py
│       ├── error.py
│       ├── io.py
│       └── utils.py
└── tests
    ├── __init__.py
    ├── test_convert.py
    └── test_version.py
```

**Comment by mitsuhiko** [comment]
I think rye should not support templates in the core of it, but I'm a strong believer in promoting good defaults. I feel like `rye init` shoudl at least also create an `src/project_name/__init__.py` file.

**Comment by con-f-use** [comment]
Yeah, this more than a brainstorm than a feature request. Whole-hearted :+1:  for `src` layout.  If `rye init` created `src/project_name/__init__.py` that is already dangerously close to a template.  Maybe provinding a `rye init --template <name>` or something and then delegating that to a configurable external command, is an option?

**Comment by mitsuhiko** [comment]
What I really like about the cargo experience is that on an empty folder it puts you on the path of success. It creates `src/lib.rs` which is exactly what you want in 99% of cases. I think the same should apply to `rye`. You _should_ be using `src/package-name`, so why not create it?

**Comment by con-f-use** [comment]
Yeah no, I absolutely meant that that should be the default behavior for `rye init` without further arguments. The `--template` would be a very optional, user/organization/project configurable type of thing in my mind.

**Comment by martin-liu** [comment]
I think tests (and coverage) config should be by default added when do `rye init`, probably also linting. 

`cargo new` don't add tests config only because it has built-in support.

**Comment by trymzet** [comment]
I initially also thought it would be nice, but then I realized running `cookiecutter gh:my-gh-org/my-cookiecutter-template-repo` is literally the same amount of effort as running `rye init --template gh:my-gh-org/my-cookiecutter-template-repo`😅

**Comment by UnoYakshi** [comment]
> ... so why bother 😅

To not keep another tool in mind.

It's not that simple to find a universal Cookiecutter template that suits one's needs. Some of the CC-templates are outdated or too poorly developed. As for Rye community, I believe it's a good chance to have way better ~content~ templates.


I always remind myself about «There should be one― and preferably only one ―obvious way to do it.» in Python Zen. But it's not applicable to how we set up Python projects. I wonder if we actually can have such templating mechanism that'd suit most of the cases. Or at least a preferred way/guidelines to start a project.
I do understand that to make all the stuff opt-in'able is kind of a headache when it comes to configurations and integrations. And that I'm basically suggesting some kind of a dev-portal from which we could download/import components in 1 click. But don't we want to at least consider something _cool_?.. Rye is THE place to experiment (or, to theorize), possibly taking dev experience to another level.
